src/plugins/intel_gna/src/gna_device.cpp

```diff
@@ -502,42 +474,42 @@ void GNADeviceHelper::dumpTLVForDeviceVersion(const uint32_t modelId,
                                               std::ostream& outStream,
                                               const std::vector<GnaEndpoint>& inputsContainer,
                                               const std::vector<GnaEndpoint>& outputsContainer) {
-    const auto compileTarget = GetCompileTarget();
     ExportTlvModel(modelId,
                    nGnaDeviceIndex,
                    outStream,
-                   compileTarget,
+                   target->get_effective_compile_target(),
                    inputsContainer,
                    outputsContainer,
                    allAllocations);
 }
 
-void GNADeviceHelper::createVirtualDevice(Gna2DeviceVersion devVersion) {
-    const auto status = Gna2DeviceCreateForExport(devVersion, &nGnaDeviceIndex);
-    GNADeviceHelper::checkGna2Status(status, "Gna2DeviceCreateForExport(" + std::to_string(devVersion) + ")");
+void GNADeviceHelper::createVirtualDevice(DeviceVersion devVersion) {
+    const auto status = Gna2DeviceCreateForExport(static_cast<Gna2DeviceVersion>(devVersion), &nGnaDeviceIndex);
+    GNADeviceHelper::checkGna2Status(status, "Gna2DeviceCreateForExport(" + DeviceToString(devVersion) + ")");
 }
 
 void GNADeviceHelper::updateGnaDeviceVersion() {
-    const auto status = Gna2DeviceGetVersion(nGnaDeviceIndex, &detectedGnaDevVersion);
+    Gna2DeviceVersion device_version;
+    const auto status = Gna2DeviceGetVersion(nGnaDeviceIndex, &device_version);
     checkGna2Status(status, "Gna2DeviceGetVersion");
+    target->set_detected_device_version(static_cast<common::DeviceVersion>(device_version));
 }
 
 void GNADeviceHelper::open() {
     std::unique_lock<std::mutex> lockGnaCalls{acrossPluginsSync};
     updateGnaDeviceVersion();
-    const auto gnaExecTarget = parseTarget(executionTarget);
-    if (useDeviceEmbeddedExport) {
-        const auto gnaCompileTarget = GetCompileTarget();
-        const auto exportGeneration = getEmbeddedTargetFromCompileTarget(gnaCompileTarget);
+    const auto execution_target = target->get_user_set_execution_target();
 
-        createVirtualDevice(exportGeneration);
+    if (useDeviceEmbeddedExport) {
+        createVirtualDevice(target->get_user_set_compile_target());
         updateGnaDeviceVersion();
-    } else if (!executionTarget.empty() && gnaExecTarget != detectedGnaDevVersion) {
-        createVirtualDevice(gnaExecTarget);
+    } else if (execution_target != DeviceVersion::NotSet && execution_target != target->get_detected_device_version()) {
+        createVirtualDevice(execution_target);
         updateGnaDeviceVersion();
-        if (detectedGnaDevVersion != gnaExecTarget) {
-            THROW_GNA_EXCEPTION << "Wrong virtual GNA device version reported: " << detectedGnaDevVersion
-                                << " instead of: " << gnaExecTarget;
+        if (target->get_detected_device_version() != execution_target) {
```

Comment:
Correct me I am wrong but there is no chance to reach this statement having condition true due the fact updateGnaDeviceVersion() will set detected_device_version of target  on success, but in case of issue it will throw.

Maybe this THROW_GNA_EXCEPTION should be moved to the updateGnaDeviceVersion method?